### ISSUE TITLE
Dirbrowser navigation problems

### ISSUE THREAD

**Issue by meles5** [The-Compiler/qutebrowser#1116]
Hi,

I guess that there's a problem with the Dirbrowser (`qutebrowser/html/dirbrowser.html`) when you want to navigate through your files: 

If you want to check out your home directory, eg. by using `:open file:home`, it will appear, but if you want to go one directory up (so to `/`, Linux command would be `cd ..`), it won't work. But if you use the command `:open file:/home`, it will work.

But where's the difference?

* If you call `:open file:home`, qutebrowser will display `Browse directory: //home/`.
* If you call `:open file:/home`, qutebrowser will display `Browse directory: /home/`.

So, my suggestion to solve this is either to disable displaying when calling `:open file:home` and mark it as an invalid url because the document root is missing, or to fix that and be able to use both variants.

**Comment by The-Compiler** [comment]
@antoyo do you have time to take a look at this?

**Comment by antoyo** [comment]
Yes, I'll look at this.
By the way, you're supposed to use `file:///home` (i.e. with 3 slashes) because `file://` is the scheme and `/home` is the path.

**Comment by antoyo** [comment]
@The-Compiler By the way, is it normal I can do `:open http:github.com` (with the missing slashes) ?

**Comment by meles5** [comment]
@antoyo Thanks, this was exactly my thought, and this is why I'm not sure if it's a bug or just a feature that shouldn't exist - I think the `/` is missing, too.

And as I understood the `http:github.com` is an abbreviation, I guess just like you're able to use `file:/home/` instead of `file:///home/`.

**Comment by The-Compiler** [comment]
Looking at [RFC 3986](https://tools.ietf.org/html/rfc3986):

```
   The generic URI syntax consists of a hierarchical sequence of
   components referred to as the scheme, authority, path, query, and
   fragment.

      URI         = scheme ":" hier-part [ "?" query ] [ "#" fragment ]

      hier-part   = "//" authority path-abempty
                  / path-absolute
                  / path-rootless
                  / path-empty

[...]


   The following are two example URIs and their component parts:

         foo://example.com:8042/over/there?name=ferret#nose
         \_/   \______________/\_________/ \_________/ \__/
          |           |            |            |        |
       scheme     authority       path        query   fragment
          |   _____________________|__
         / \ /                        \
         urn:example:animal:ferret:nose

[...]

      path          = path-abempty    ; begins with "/" or is empty
                    / path-absolute   ; begins with "/" but not "//"
                    / path-noscheme   ; begins with a non-colon segment
                    / path-rootless   ; begins with a segment
                    / path-empty      ; zero characters
```

Looking at how `QUrl` behaves:

```
>>> u = QUrl('file:home'); (u.scheme(), u.host(), u.path())
('file', '', 'home')
>>> u = QUrl('file:/home'); (u.scheme(), u.host(), u.path())
('file', '', '/home')
>>> u = QUrl('file:///home'); (u.scheme(), u.host(), u.path())
('file', '', '/home')
>>> u = QUrl('http:github.com'); (u.scheme(), u.host(), u.path())
('http', '', 'github.com')
>>> u = QUrl.fromUserInput('http:github.com'); (u.scheme(), u.host(), u.path())
('http', '', 'github.com')
```

Looking at that I think the behaviour of `file:home` (`path-rootless`), `file:/home` (`path-absolute`) and `file:///home` (`path-abempty`) makes sense.

I don't get why `http:github.com` works though, probably some magic happening inside QtWebKit.

**Comment by antoyo** [comment]
Does this mean the `path-rootless` is a relative path?
So `file:home` should show the `home` directory in the current working directory?

**Comment by Kingdread** [comment]
I think the base issue is fixed in 375e60627a69d96f4743842c93ca1871a50383c7, at least it seems like I can `:open file:home`, click `..` and land in the listing for `/`.

So I think we can close this, unless we want to change this:

> So file:home should show the home directory in the current working directory?

**Comment by The-Compiler** [comment]
Oh, sorry for not answering here @antoyo, I must've forgotten about it somehow.

I don't think being able to open a path relative to qutebrowser's invocation path is an useful thing to do, so I think the current behavior is fine.
